xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/job/question/XClassBreakingQuestion.data.vm

```diff
@@ -0,0 +1,163 @@
+###
+### Return the list of pages and extensions that are going to be deleted by a delete job
+###
+$response.setContentType("application/json")
+
+###
+### Globals
+###
+#set ($offset = "$!request.offset")
+#if ($offset == '')
+  #set ($offset = '0')
+#end
+#set ($offset = $numbertool.toNumber($offset))
+#set ($limit = 15)
+
+#**
+ * Display a pagination node.
+ * @param $data the collection where the node will be added
+ * @param $howMany the number of nodes that remains
+ * @param $newOffset the offset to use to display the next batch of nodes
+ *#
+#macro(paginationNode $data $howMany $newOffset)
+  #set ($id = "$!{request.id}")
+  #if ($id == '')
+    #set ($id = '#')
+  #end
+  #set ($discard = $data.add({
+    'id': "pagination:$id",
+    'text': "$escapetool.javascript($services.localization.render('job.question.XClassBreakingQuestion.tree.paginationNode', [$howMany]))",
+    'icon': 'fa fa-eye',
+    'children': false,
+    'data': {
+    'type': 'pagination',
+    'validChildren': [],
+    'canDelete': true,
+    'offset': $newOffset
+  },
+    'li_attr': {
+    'class': 'paginationNode'
+  }
+  }))
+#end
+
+#macro(displayFreePages $question $data)
+  #set ($pages = $question.freePages)
+  #foreach ($page in $pages)
+    #set ($pageReference = $page.entityReference)
+    #if ($velocityCount > $offset + $limit)
+      #set ($howMany = $pages.size() - $velocityCount + 1)
+      #paginationNode($data, $howMany, $velocityCount)
+      #break
+    #end
+    #if ($velocityCount > $offset)
+      #set ($pageDoc = $xwiki.getDocument($pageReference))
+      #set ($title = $pageDoc.getRenderedTitle('html/5.0'))
```

Comment:
getDisplayTitle() will find the target syntax automatically (which might be another variant of html).

Reply:
Actually the Javadoc says that `getDisplayTitle()` is deprecated and that I should use `getRenderedTitle()` instead, see: https://github.com/xwiki/xwiki-platform/blob/6fd9baa341e370206744f1577641316177e3d2e4/xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-oldcore/src/main/aspect/com/xpn/xwiki/doc/XWikiDocumentCompatibilityAspect.aj#L162-L172